wikidict/render.py

```diff
@@ -0,0 +1,386 @@
+    return files[-1]
+
+
+def load(input_file: Path) -> Dict[str, str]:
+    """Load the big JSON file containing all words and their details."""
+    with input_file.open(encoding="utf-8") as fh:
+        words: Dict[str, str] = json.load(fh)
+    print(f">>> Loaded {len(words):,} words from {input_file}", flush=True)
+    return words
+
+
+def debug_render(in_words: Dict[str, str], locale: str) -> None:
+    sections = defaultdict(list)
+    templates: Dict[str, str] = {}
+    for in_word, code in in_words.items():
+        for title in find_titles(code, locale):
+            sections[title].append(in_word)
+
+        parsed_sections = find_sections(code, locale)
+        defs = "\n".join(str(s) for s in parsed_sections.values())
+        for template in re.findall(r"({{[^{}]*}})", defs):
+            template = template.split("|")[0].lstrip("{").rstrip("}").strip()
+            if template not in templates:
+                templates[template] = in_word
+
+    with open("sections.txt", "w") as f:
+        for title, entries in sorted(sections.items()):
+            f.write(f"{title!r} ({len(entries):,})\n")
+            if len(entries) < 10:
+                # Most likely errors/mispellings
+                for entry in entries:
+                    f.write(f"    - {entry!r}\n")
+            else:
+                f.write(f"    - {entries[0]!r}\n")
+
+    if templates:
+        with open("templates.txt", "w") as f:
+            for template, entry in sorted(templates.items()):
+                f.write(f"{entry!r} => {template!r}\n")
+
+
+def render(in_words: Dict[str, str], locale: str) -> Words:
+    words: Words = {}
+    for in_word, code in in_words.items():
+        try:
+            details = parse_word(in_word, code, locale)
+        except Exception:  # pragma: nocover
+            print(f"ERROR with {in_word!r}", flush=True)
+        else:
+            if details.definitions or details.variants:
+                words[in_word] = details
+    return words
+
+
+def save(snapshot: str, words: Words, output_dir: Path) -> None:
+    """Persist data."""
+    # This file is needed by convert.py
+    raw_data = output_dir / f"data-{snapshot}.json"
+    with raw_data.open(mode="w", encoding="utf-8") as fh:
+        json.dump(words, fh, indent=4, sort_keys=True)
+
+    (output_dir / "words.count").write_text(str(len(words)))
+    (output_dir / "words.snapshot").write_text(snapshot)
+
+    print(f">>> Saved {len(words):,} words into {raw_data}", flush=True)
+
+
+def main(locale: str) -> int:
+    output_dir = Path(os.getenv("CWD", "")) / "data" / locale
+    filename = get_latest_json_file(locale, output_dir)
+    if not filename:
+        print(">>> No data-XXXXXX.json found. Run with --parse first ... ", flush=True)
+        return 1
+
+    current_file = Path(output_dir, filename)
+
+    print(f">>> Loading {current_file} ...")
+    in_words: Dict[str, str] = load(current_file)
+
+    debug = "DEBUG" in os.environ
+    if debug:
```

Comment:
```suggestion
    if "DEBUG" in os.environ:
```

Reply:
I will fix this, but I would suggest to create a special --option for this. --find-sections-templates ?

Reply:
Or simply `--find-templates`? But yes, a specific argument would be better.

Reply:
Ok done. find-templates is now an option, so is --get-word.
wikidict is now on par with scripts.get